File: sdk/kotlin-enhanced/src/commonMain/kotlin/com/flagent/enhanced/entry/FlagentManagerAdapter.kt
package com.flagent.enhanced.entry

import com.flagent.client.models.EvalResult
import com.flagent.client.models.EvaluationEntity
import com.flagent.enhanced.manager.FlagentManager

/**
 * Adapts [FlagentManager] (server-side) to [FlagentClient].
 */
class FlagentManagerAdapter(
    private val manager: FlagentManager
) : FlagentClient {

    override suspend fun evaluate(
        flagKey: String?,
        flagID: Long?,
        entityID: String?,
        entityType: String?,
        entityContext: Map<String, Any>?,
        enableDebug: Boolean
    ): EvalResult = manager.evaluate(
        flagKey = flagKey,
        flagID = flagID,
        entityID = entityID,
        entityType = entityType,
        entityContext = entityContext,
        enableDebug = enableDebug
    )

    override suspend fun evaluateBatch(
        flagKeys: List<String>?,
        flagIDs: List<Int>?,
        entities: List<EvaluationEntity>,
        enableDebug: Boolean
    ): List<EvalResult> = manager.evaluateBatch(
        flagKeys = flagKeys,
        flagIDs = flagIDs,
        entities = entities,
        enableDebug = enableDebug
    )
}

File: sdk/swift-enhanced/Sources/FlagentEnhanced/Client/FlagentManagerAdapter.swift
import Foundation
import FlagentClient
import AnyCodable

/// Adapts `FlagentManager` to the unified `FlagentClient` protocol and adds `isEnabled`.
@available(macOS 10.15, iOS 13.0, *)
public final class FlagentManagerAdapter: FlagentClient {
    private let evaluator: FlagentEvaluating

    public init(manager: FlagentManager) {
        self.evaluator = manager
    }

    init(evaluator: FlagentEvaluating) {
        self.evaluator = evaluator
    }

    public func evaluate(
        flagKey: String? = nil,
        flagID: Int64? = nil,
        entityID: String? = nil,
        entityType: String? = nil,
        entityContext: [String: AnyCodable]? = nil,
        enableDebug: Bool = false
    ) async throws -> EvalResult {
        try await evaluator.evaluate(
            flagKey: flagKey,
            flagID: flagID,
            entityID: entityID,
            entityType: entityType,
            entityContext: entityContext,
            enableDebug: enableDebug
        )
    }

    public func isEnabled(
        flagKey: String,
        entityID: String? = nil,
        entityType: String? = nil,
        entityContext: [String: AnyCodable]? = nil
    ) async throws -> Bool {
        let result = try await evaluator.evaluate(
            flagKey: flagKey,
            flagID: nil,
            entityID: entityID,
            entityType: entityType,
            entityContext: entityContext,
            enableDebug: false
        )
        return result.variantKey != nil
    }

    public func evaluateBatch(
        flagKeys: [String]? = nil,
        flagIDs: [Int64]? = nil,
        entities: [EvaluationEntity],
        enableDebug: Bool = false
    ) async throws -> [EvalResult] {
        try await evaluator.evaluateBatch(
            flagKeys: flagKeys,
            flagIDs: flagIDs,
            entities: entities,
            enableDebug: enableDebug
        )
    }
}
